source/docs/software/basic-programming/java-gc.rst

```diff
@@ -0,0 +1,33 @@
+Java Garbage Collection
+=======================
```

Comment:
I'm not a fan of placing this under basics, even with it being the catchall that it is. I wonder if it makes more sense to create a subsection called language concepts.

Reply:
It snuck into the basics article for Java Units, so it is here now. I agree it didn't belong here. The other reviewers feel it's critical information when dealing with the units library, for some reason, even though WPILib has several other core immutable types that are designed to be recreated on every period, such as `Pose2d` and `ModuleStates`.

Maybe it should be removed from this PR (and the Java Units library documentation), and I can write up a separate issue for it.

Reply:
> The other reviewers feel it's critical information when dealing with the units library, for some reason, even though WPILib has several other core immutable types that are designed to be recreated on every period, such as Pose2d and ModuleStates.

For typical math, unit containers get instantiated _way_ more than any of our other value types.

Reply:
> For typical robot code math, unit container value types would get instantiated way more than any of our other value types

It's beyond the scope of this document PR, but this comment makes me question if there should be arithmetic functions that return Measures on the Units library at all, if object instantiation is such an issue. Remember, our target audience is 13-18 year old students with little to no Java experience. Troubleshooting the problem starting on line 245 in this doc will be more difficult for a student or CSA than debugging a classic unit error. This feature just swaps one problem that takes careful consideration (units) for another (object ownership and mutability.)

It might be best to only offer a `replace()` mutation function, and give guidance on appropriate use of the Units library. Another option would be to only offer arithmetic functions where the called must pass a unit and it returns double, to avoid object instantiation, like `double add(double, Unit)` and `double add(Measure<Unit>, Unit)`.

To bring this back on topic, should I leave the java-gc page in the Basics section, or remove it and create a new PR?

Reply:
Maybe we should rename basic programming to miscellaneous. It's really a grab bag of topics that aren't really basic programming

Reply:
So I should I remove the article and create a new issue for this debate?